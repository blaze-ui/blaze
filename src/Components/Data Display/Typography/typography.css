@import url("https://fonts.googleapis.com/css?family=Montserrat&display=swap");
.typography {
  font-family: "Montserrat", sans-serif;
}
/*# sourceMappingURL=typography.css.map */