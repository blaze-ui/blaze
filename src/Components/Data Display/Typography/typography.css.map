{
    "version": 3,
    "mappings": "AAAA,OAAO,CAAC,sEAAI;AAEZ,AAAA,WAAW,CAAC;EACV,WAAW,EAAE,wBAAwB;CACtC",
    "sources": [
        "typography.scss"
    ],
    "names": [],
    "file": "typography.css"
}