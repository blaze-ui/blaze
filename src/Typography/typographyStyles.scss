@import "../Styles/globals.scss";

.typography {
  font-family: $zapFont, sans-serif;
}
