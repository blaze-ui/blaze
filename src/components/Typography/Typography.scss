@import "../../styles/ColorSchema.scss";

.typography {
  font-family: $zapFont, sans-serif;
  color: silver;
}
