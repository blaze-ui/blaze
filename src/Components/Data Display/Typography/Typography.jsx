import React, { Component } from "react";
import "./typography.scss";

export default class Typography extends Component {
  render() {
    return <div className="typography">{this.props.children}</div>;
  }
}
