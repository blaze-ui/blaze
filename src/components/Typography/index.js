import React from "react";
import PropTypes from "prop-types";
import "./Typography.css";

function Typography(props) {
  const { variant, children, color } = props;
  const _styles = { color };
  switch (variant) {
    case "h1":
      return (
        <h1 className="typography" style={_styles}>
          {children}
        </h1>
      );
    case "h2":
      return (
        <h2 className="typography" style={_styles}>
          {children}
        </h2>
      );
    case "h3":
      return (
        <h3 className="typography" style={_styles}>
          {children}
        </h3>
      );
    case "h4":
      return (
        <h4 className="typography" style={_styles}>
          {children}
        </h4>
      );
    case "h5":
      return (
        <h5 className="typography" style={_styles}>
          {children}
        </h5>
      );
    default:
      return (
        <p className="typography" style={_styles}>
          {children}
        </p>
      );
  }
}
Typography.propTypes = {
  variant: PropTypes.string,
  children: PropTypes.string,
  color: PropTypes.string
};

Typography.defaultProps = {
  variant: "h5",
  children: "Zapify Ui Text",
  color: "#111"
};
export default Typography;
