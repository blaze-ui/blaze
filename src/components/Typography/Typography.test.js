import React from "react";
import renderer from "react-test-renderer";
import Typography from "./index";

test("Container tests", () => {
  const component = renderer.create(
    <Typography variant="h5">Zapify Ui Text</Typography>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
