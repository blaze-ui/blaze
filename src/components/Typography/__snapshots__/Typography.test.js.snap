// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container tests 1`] = `
<h5
  className="typography"
>
  Zapify Ui Text
</h5>
`;
